shell/common/shell_unittests.cc

```diff
@@ -1217,6 +1217,55 @@ TEST_F(ShellTest,
   DestroyShell(std::move(shell), std::move(task_runners));
 }
 
+TEST_F(ShellTest, GetUsedThisFrameShouldBeSetBeforeEndFrame) {
+  auto settings = CreateSettingsForFixture();
+  fml::AutoResetWaitableEvent end_frame_latch;
+  std::shared_ptr<ShellTestExternalViewEmbedder> external_view_embedder;
+  bool used_this_frame = true;
+  auto end_frame_callback =
+      [&](bool should_resubmit_frame,
+          fml::RefPtr<fml::RasterThreadMerger> raster_thread_merger) {
+        end_frame_latch.Signal();
```

Comment:
Shouldn't this go after `used_this_frame = ...` ?

Reply:
It absolutely should! Thanks!

Reply:
https://github.com/flutter/engine/pull/33649